java/src/main/c++/vw_learner_VWFactory.cc

```diff
@@ -0,0 +1,51 @@
+#include "../../../../vowpalwabbit/vw.h"
+#include "jni_base_learner.h"
+#include "vw_learner_VWFactory.h"
+
+#define RETURN_TYPE "vw/learner/VWFactory$VWReturnType"
+#define RETURN_TYPE_INSTANCE "L" RETURN_TYPE ";"
+
+JNIEXPORT jlong JNICALL Java_vw_learner_VWFactory_initialize(JNIEnv *env, jobject obj, jstring command)
+{ jlong vwPtr = 0;
+  try
+  { vw* vwInstance = VW::initialize(env->GetStringUTFChars(command, NULL));
+    vwPtr = (jlong)vwInstance;
+  }
+  catch(...)
+  { rethrow_cpp_exception_as_java_exception(env);
+  }
+  return vwPtr;
+}
+
+JNIEXPORT void JNICALL Java_vw_learner_VWFactory_closeInstance(JNIEnv *env, jobject obj, jlong vwPtr)
+{ try
+  { VW::finish(*((vw*)vwPtr));
+  }
+  catch(...)
+  { rethrow_cpp_exception_as_java_exception(env);
+  }
+}
+
+JNIEXPORT jobject JNICALL Java_vw_learner_VWFactory_getReturnType(JNIEnv *env, jobject obj, jlong vwPtr)
+{
+    jclass clVWReturnType = env->FindClass(RETURN_TYPE);
+    jfieldID field;
+    vw* vwInstance = (vw*)vwPtr;
+    if (vwInstance->p->lp.parse_label == simple_label.parse_label) {
+        if (vwInstance->lda > 0)
+            field = env->GetStaticFieldID(clVWReturnType , "VWFloatArrayType", RETURN_TYPE_INSTANCE);
+        else
+            field = env->GetStaticFieldID(clVWReturnType , "VWFloatType", RETURN_TYPE_INSTANCE);
+    }
+    else if (vwInstance->p->lp.parse_label == MULTILABEL::multilabel.parse_label)
+        field = env->GetStaticFieldID(clVWReturnType , "VWIntArrayType", RETURN_TYPE_INSTANCE);
+    else if (vwInstance->p->lp.parse_label == MULTICLASS::mc_label.parse_label ||
+             vwInstance->p->lp.parse_label == CB::cb_label.parse_label ||
```

Comment:
One minor note here is that I'm not sure what the difference between CB and CB_EVAL is.